pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -149,6 +149,43 @@ def test_collection_field(self):
             'collection names not proper'
         )
 
+    def test_search_errata(self):
+        """Test whether ``updateinfo.xml`` has errata with modular information.
+
+        This Test does the following:
+
+        1. Create and Sync a repo with ``RPM_WITH_MODULES_FEED_URL``.
+        2. Check whether ``updateinfo.xml`` has an errata with modules.
+        3. Check whether search api returns modular content.
+
+        This test case targets:
+
+        * `Pulp #4112 <https://pulp.plan.io/issues/4112>`_.
+        """
+        # Step 1
+        body = gen_repo(
+            importer_config={'feed': RPM_WITH_MODULES_FEED_URL},
+            distributors=[gen_distributor(auto_publish=True)]
+        )
+        repo = self.client.post(REPOSITORY_PATH, body)
+        sync_repo(self.cfg, repo)
+        self.addCleanup(self.client.delete, repo['_href'])
+        repo = self.client.get(repo['_href'], params={'details': True})
+        # Step 2
+        update_info_file = get_repodata(
+            self.cfg,
+            repo['distributors'][0],
+            'updateinfo'
+        )
+        self.assertTrue(update_info_file.findall('.//module'))
```

Comment:
What about we make those assertions a bit more strict? For instance:
Assert that number of module units are per the updateinfo.xml/
Assert that name field, stream, version, context are not empty.